general/q3.py

```diff
@@ -0,0 +1,25 @@
+def secondSmallestInArray(array):
+    if len(array) < 2:
+        raise ValueError
+    
+    smallest = None
+    secondSmallest = None
+
+    for i in array:
+        if smallest == None:
+            smallest = i
+
+        elif i < smallest:
+            secondSmallest = smallest
+            smallest = i
+
+        elif secondSmallest == None:
+            secondSmallest = i
+        
+        elif i > smallest and i < secondSmallest:
```

Comment:
Works but quite confusing to read 

something simple like sorted(set(array))[1] would work

Reply:
also quicker alternatives like:
[*set(input_list.sort())][1]